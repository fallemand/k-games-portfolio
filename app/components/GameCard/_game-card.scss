@import '../../styles/variables';
@import 'game-card__small';

@media ( min-width: $breakpoint-desktop ) {
  @import 'game-card__medium';
}

