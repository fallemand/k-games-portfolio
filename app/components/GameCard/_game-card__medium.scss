.game-card {
  width: 500px;
  height: 220px;

  &:nth-child(odd) {
    margin-right: 40px;
  }

  .game-card__image {
    height: 100%;
    width: auto;
    transition: filter $transition;
  }

  .game-card__description {
    padding: 10px 20px;
    height: 160px;
    transform: translate3d(0, -50px, 0);
    transition: all $transition;
  }

  .game-card__button {
    width: 150px;
    margin-right: 10px;
  }

  &:hover {
    box-shadow: 0 6px 10px 0 rgba( 0, 0, 0, 0.3);

    .game-card__description {
      transform: translate3d(0, -110px, 0);
      z-index: 100;
      background-color: $card-background--color-hover;
    }
    .game-card__image {
      filter: blur(3px);
    }
  }
}
