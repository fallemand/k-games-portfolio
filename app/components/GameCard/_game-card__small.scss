$card-background-color: rgba(255, 255, 255, 0.75);
$card-background--color-hover: rgba(255, 255, 255, 0.95);

.game-card {
  box-shadow: 0 2px 4px 0 rgba(0 , 0, 0, 0.3);
  border-radius: $border-radius;
  width: 100%;
  margin: 10px 0;
  display: inline-block;
  overflow: hidden;
  transition: box-shadow $transition;

  .game-card__image {
    height: auto;
    width: 100%;
    z-index: 10;
    transition: filter $transition;
  }

  .game-card__description {
    padding: 10px;
    background-color: $card-background-color;
  }

  .game-card__title {
    margin: 0;
    font-weight: $font-bold;
    font-size: $font-size-22;
    color: $font-color-primary;
  }

  .game-card__actions {
    margin-top: 10px;
  }

  .game-card__button {
    width: 46%;
    margin: 0 2%;
  }
}
