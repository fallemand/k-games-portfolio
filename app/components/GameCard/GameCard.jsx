import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Button from '../Button';
import './_game-card.scss';

const GameCard = ({ className, game, onAction, actions }) => (
  <div className={classnames(className, 'game-card')}>
    <img className="game-card__image" src={game.images.banner} alt={`${game.name}`} />
    <div className="game-card__description">
      <h3 className="game-card__title">{game.name}</h3>
      <div className="game-card__actions">
        {actions.includes('add') && (
          <Button onClick={() => onAction('add', game.short)} color={Button.colors.GREEN}>
            Add to Portfolio
          </Button>
        )}
        {actions.includes('remove') && (
          <Button onClick={() => onAction('remove', game.short)} color={Button.colors.DEFAULT}>
            Remove from Portfolio
          </Button>
        )}
        {actions.includes('play') && (
          <Button onClick={() => onAction('play', game.short)} color={Button.colors.BLUE}>
            Play
          </Button>
        )}
      </div>
    </div>
  </div>
);

GameCard.propTypes = {
  className: PropTypes.string,
  game: PropTypes.object.isRequired,
  onAction: PropTypes.func.isRequired,
  actions: PropTypes.array.isRequired,
};

GameCard.defaultProps = {
  className: null,
};

export default GameCard;
