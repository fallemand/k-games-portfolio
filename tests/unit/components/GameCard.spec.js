import React from 'react';
import GameCard from '../../../app/components/GameCard';

describe('GameCard', () => {
  const props = {
    className: '__CLASSNAME__',
    game: {
      name: '__NAME__',
      short: '__ID__',
      images: {
        banner: '__BANNERURL__',
      },
    },
    onAction: jest.fn(),
    actions: ['play'],
  };

  it('render with props', () => {
    const component = shallow(<GameCard {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('should change buttons according to actions', () => {
    const component = shallow(<GameCard {...props} actions={['remove', 'play']} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
    component.setProps({ actions: ['add', 'play'] });
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('click on button should call onAction', () => {
    const component = shallow(<GameCard {...props} />);
    component.find('Button').simulate('click');
    expect(props.onAction).toHaveBeenCalledWith('play', '__ID__');
  });
});
