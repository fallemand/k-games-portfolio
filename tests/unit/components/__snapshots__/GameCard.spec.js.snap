// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard render with props 1`] = `
"<div className='__CLASSNAME__ game-card'>
  <img className='game-card__image' src='__BANNERURL__' alt='__NAME__' />
  <div className='game-card__description'>
    <h3 className='game-card__title'>
      __NAME__
    </h3>
    <div className='game-card__actions'>
      <Button onClick={[Function: onClick]} color='blue' className='game-card__button'>
        Play
      </Button>
    </div>
  </div>
</div>"
`;

exports[`GameCard should change buttons according to actions 1`] = `
"<div className='__CLASSNAME__ game-card'>
  <img className='game-card__image' src='__BANNERURL__' alt='__NAME__' />
  <div className='game-card__description'>
    <h3 className='game-card__title'>
      __NAME__
    </h3>
    <div className='game-card__actions'>
      <Button onClick={[Function: onClick]} color='default' className='game-card__button'>
        Remove
      </Button>
      <Button onClick={[Function: onClick]} color='blue' className='game-card__button'>
        Play
      </Button>
    </div>
  </div>
</div>"
`;

exports[`GameCard should change buttons according to actions 2`] = `
"<div className='__CLASSNAME__ game-card'>
  <img className='game-card__image' src='__BANNERURL__' alt='__NAME__' />
  <div className='game-card__description'>
    <h3 className='game-card__title'>
      __NAME__
    </h3>
    <div className='game-card__actions'>
      <Button onClick={[Function: onClick]} color='green' className='game-card__button'>
        Add to Portfolio
      </Button>
      <Button onClick={[Function: onClick]} color='blue' className='game-card__button'>
        Play
      </Button>
    </div>
  </div>
</div>"
`;
